pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -120,8 +120,11 @@ func SetNodeGroupDefaults(ng *NodeGroup, meta *ClusterMeta, controlPlaneOnOutpos
 // SetManagedNodeGroupDefaults sets default values for a ManagedNodeGroup
 func SetManagedNodeGroupDefaults(ng *ManagedNodeGroup, meta *ClusterMeta, controlPlaneOnOutposts bool) {
 	setNodeGroupBaseDefaults(ng.NodeGroupBase, meta)
-	if ng.AMIFamily == "" {
-		ng.AMIFamily = NodeImageFamilyAmazonLinux2
+
+	// When using custom AMIs, we want the user to explicitly specify AMI family.
+	// Thus, we only setup default AMI family when no custom AMI is being used.
+	if ng.AMIFamily == "" && ng.AMI == "" {
+		ng.AMIFamily = DefaultNodeImageFamily
```

Comment:
Shouldn't these changes be for line 103?

Reply:
Self managed nodegroup default values are being set **_after_** validations, thus the validations are being done directly on user input and we don't need these changes there.
![image](https://user-images.githubusercontent.com/110664232/217242953-573c83d5-bf30-4ca0-9416-6c5fd5079f20.png)

For EKS managed nodegroups, the default values are being set _**before**_ validations. Thus, we don't want to set the default AMI Family before validating, as we want the user to do it explicitly.
![image](https://user-images.githubusercontent.com/110664232/217243435-74eca58f-436b-4550-896b-e64b186946d7.png)